scripts/docker-run.sh

```diff
@@ -38,6 +38,27 @@ else
 	DOCKER_RUN="docker run"
 fi
 
+#
+# Ensure that either UPSTREAM_PRODUCT_BRANCH or DELPHIX_PACKAGE_MIRROR_MAIN
+# is set. If neither is set, attempt to use git to determine the upstream
+# branch. If that also fails, then exit.
+#
+UPSTREAM_BRANCH="$UPSTREAM_PRODUCT_BRANCH"
+if [[ -z "$UPSTREAM_BRANCH" ]]; then
+    echo "UPSTREAM_PRODUCT_BRANCH is not set."
+    echo "Attempting to infer UPSTREAM_BRANCH."
+    UPSTREAM_BRANCH=$(git for-each-ref --format='%(upstream:short)' \
```

Comment:
I think @prakashsurya mentioned that since this will not be run on a developer laptop, the upstream branch will likely not be set. Also, another reason why using an upstream is likely problematic: let's say I clone my private appliance-build repository, which has the branch "testing" to which I just pushed from my laptop. If I then do `git checkout testing`, it will create a new branch named "testing" which tracks "origin/testing", and so UPSTREAM_PRODUCT_BRANCH will be set to "testing", which is definitely not what we want.

I'd propose we just fail and mention a message such as:
```please set UPSTREAM_PRODUCT_BRANCH to the version of the product branch you are trying to build (e.g. "master", "6.0/stage", "release/6.0.0.0")```. By giving concrete examples of product branches, the developer is much more likely to provide the correct value.

Reply:
Given what @prakashsurya mentioned I am leaning more towards including the hard coded BRANCH parameter in the repository itself, and having that updated by the release lead. I think that will make it easy to use for developers, and difficult for them to screw up the configuration.

Reply:
After talking with @pzakha offline I opted to ensure that UPSTREAM_PRODUCT_BRANCH is always set. If this ends up being too cumbersome for developers I will change this in the future.

Reply:
Can a summary of that discussion be posted here, so that all parties involved can have a say in the decision.

Given that we’ve been going back and forth on this issue, I think it’s only appropriate to have that discussion in the the open on this PR.

Additionally it’s good to have that discussion captured here in case the decision comes into scrutiny at some later time.

I think it’s perfectly reasonable to use a hard-coded value. Especially given, when we create a new release branch(es) we’ll have to create the new mirrors. So it’s a natural extension that after we create the mirror for the new release, we explicitly update the code to start using that new mirror.

If we allow the value to be passed it, it opens the door to be incorrectly specified, which I’m not sure how that error would get handled; e.g. if the user specifies “master” when building an older release.

Reply:
The summary is that this is the easiest way to go forward for now to get this work integrated, but that we will most likely want to get the branches hard-coded in the longer term. However, we should spend some time to do it properly, both for this repository and for linux-pkg. I've created this issue to track the future work: https://jira.delphix.com/browse/LB-221.

Another question is once we hardcode this, what do we do with the value currently passed in UPSTREAM_PRODUCT_BRANCH through Jenkins? Ideally I think we should remove it as there will be 2 sources of truth which would be confusing.

Reply:
What do you mean by do it properly? How was the prior solution proposed not proper? Since the discussion where you swayed @grodr was not done here in the open, it’s hard for me to understand what you mean. 

I do not agree that this is the easiest way to go forward, and I do not agree that this is the best solution in general.

But clearly you are not going to try to drive to a consensus here, but rather you want your approach taken and opt for offline conversations to keep me out of the loop to achieve that goal.

I don’t have the patience, so do what you will, but please don’t count me as a reviewer.